src/stores/manager/keys/local.go

```diff
@@ -55,5 +55,5 @@ func NewLocalKeyStore(specs *LocalKeySpecs, db database.Secrets, logger log.Logg
 		return nil, err
 	}
 
-	return localkeys.New(secretStore, logger), nil
+	return localkeys.New(secretStore, db, logger), nil
```

Comment:
I don't think this is how this should be implemented. IMO the local keys should accept a secret connector as dependency instead of a secret store. The secret connector already has all the functions you implemented